Utilities/Doxygen/DoxygenConfig.cmake

```diff
@@ -0,0 +1,163 @@
+# Configure cmake variables used to autogenerate a doxyfile configuration
```

Comment:
There is another `DoxygenConfig.cmake` in Wrapping directory. Is the duplication/redundancy intentional?

Reply:
It is intentional.  Much of it is duplicated, and it could be consolidated with a little cmake logic.  

The confound is that one is needed for Wrapping and the other is needed in Utilities/Documentation.  

In order to support ITK_WRAP_DOC from external builds, the Wrapping version should stay there.